src/main/java/org/opensearch/flowframework/workflow/ReindexStep.java

```diff
@@ -95,10 +95,20 @@ public PlainActionFuture<WorkflowData> execute(
             Float requestsPerSecond = inputs.containsKey(REQUESTS_PER_SECOND)
                 ? Float.parseFloat(inputs.get(REQUESTS_PER_SECOND).toString())
                 : null;
+            requestsPerSecond = requestsPerSecond < 0 ? Float.POSITIVE_INFINITY : requestsPerSecond;
             Boolean requireAlias = inputs.containsKey(REQUIRE_ALIAS) ? Booleans.parseBoolean(inputs.get(REQUIRE_ALIAS).toString()) : null;
-            Integer slices = (Integer) inputs.get(SLICES);
-            Integer maxDocs = (Integer) inputs.get(MAX_DOCS);
-
+            Integer slices;
+            Integer maxDocs;
+            if (inputs.get(SLICES) != null) {
+                slices = Integer.parseInt(String.valueOf(inputs.get(SLICES)));
+            } else {
+                slices = (Integer) inputs.get(SLICES);
+            }
```

Comment:
```java
slices = inputs.get(SLICES) instanceof String
         ? Integer.parseInt(String.valueOf(inputs.get(SLICES)))
         : (Integer) inputs.get(SLICES);
```

Reply:
I need to a != null check and with ternary operator you can't do that cause you need both sides of : to not be null unlike if/else